code/messaging/src/test/java/com/adobe/marketing/mobile/MessagingTests.java

```diff
@@ -466,6 +499,136 @@ public void test_handleNotificationResponseWithEmptyAction() {
+        when(mockIntent.getStringExtra("messageId")).thenReturn("mockMessageId");
+        when(mockIntent.getStringExtra("adobe_xdm")).thenReturn("mockXdmData");
+        when(mockIntent.getStringExtra("adb_iam_id")).thenReturn("mockInappId");
+
+        final ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+        runWithMockedMobileCore(
+                eventCaptor,
+                null,
+                () -> {
+                    try (MockedStatic<Log> logMockedStatic =
+                                    Mockito.mockStatic(Log.class, Mockito.CALLS_REAL_METHODS);
+                            MockedStatic<Messaging> messagingMockedStatic =
+                                    Mockito.mockStatic(
+                                            Messaging.class, Mockito.CALLS_REAL_METHODS)) {
+
+                        messagingMockedStatic
+                                .when(
+                                        () ->
+                                                Messaging.updatePropositionsForSurfaces(
+                                                        anyList(), any(AdobeCallback.class)))
+                                .thenAnswer(
+                                        (Answer<Void>)
+                                                invocation -> {
+                                                    AdobeCallback<Boolean> booleanAdobeCallback =
+                                                            invocation.getArgument(1);
+                                                    booleanAdobeCallback.call(false);
+                                                    return null;
+                                                });
+
+                        Messaging.handleNotificationResponse(mockIntent, true, "mockActionId");
+
+                        Event messagingEvent = eventCaptor.getAllValues().get(1);
+                        assertEquals(
+                                "Push notification interaction event", messagingEvent.getName());
+                        assertEquals(
+                                "com.adobe.eventSource.requestContent", messagingEvent.getSource());
+                        assertEquals("com.adobe.eventType.messaging", messagingEvent.getType());
+                        assertEquals(
+                                "mockMessageId", messagingEvent.getEventData().get("messageId"));
+                        assertEquals("mockActionId", messagingEvent.getEventData().get("actionId"));
+                        assertEquals(true, messagingEvent.getEventData().get("applicationOpened"));
+                        assertEquals("mockXdmData", messagingEvent.getEventData().get("adobe_xdm"));
+                        assertEquals(
+                                "pushTracking.customAction",
+                                messagingEvent.getEventData().get("eventType"));
+
+                        Event pushToInappEvent = eventCaptor.getAllValues().get(0);
```

Comment:
Should this even be dispatched if the `updatePropositionsForSurfaces` callback returns false?